### ISSUE TITLE
Maven artifacts repo

### ISSUE THREAD

**Issue by ebremer** [jcefbuild/java-cef-build#9]
Is there a maven repo to provide platform artifacts for use with maven builds?

**Comment by smac89** [comment]
Not at the moment, but I hope to be able to do something about this soon.

We should be able to provide packages right here from github or jfrog. I don't know how this will work and whether or not I am allowed to do it, but it is something I have thought of doing in the past

**Comment by guusdk** [comment]
I believe you can use Github itself as a Maven repository: https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-apache-maven-for-use-with-github-packages

**Comment by RutenkolkC** [comment]
It seems like you would need an access token to install github packages from maven. That would be very inconvenient to use this dependency in any CI.

A proper maven artifact repository would also help using jcef in other jvm languages like kotlin or clojure and build tools like gradle and leiningen.

Not having a proper maven dependency is what hold me back from looking more closely into jcef in the past. I simply thought jcef wasnt done, or stable yet. Only Jetbrains recent announcement to move from javafx to jcef sparked another "oh, I guess there must finally be maven deps then".

**Comment by smac89** [comment]
This seems to have picked

> I believe you can use Github itself as a Maven repository: https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-apache-maven-for-use-with-github-packages

@guusdk Yes, I've used it earlier this year it is still alpha at best. It just didn't behave like a maven repo. When I start work on #7 I will see if there has been any improvement

**Comment by FriwiDev** [comment]
Im currently working on maven support with automatic builds. The natives are downloaded from this repository. I currently have support for Linux and Windows ready, I plan on creating a custom bundle plugin for all platforms (including MacOS) on top of that soon. The current version extracts the natives from the classpath, which is great for testing on the supported platforms. On MacOS this option is not available, as CEF requires a bundle on MacOS. I'll keep you guys updated on any progress I make.

**Comment by FriwiDev** [comment]
Maven artifacts are now available: https://github.com/jcefmaven/jcefmaven

**Comment by supermoonie** [comment]
@FriwiDev  Thanks ! It worked ! But it only work with java8  ?

**Comment by FriwiDev** [comment]
@supermoonie I did in fact not test the setup with Java 8. If you run into any issues using newer versions, feel free to create an issue so that I can look into it :)

**Comment by supermoonie** [comment]
![image](https://user-images.githubusercontent.com/17509860/101233200-d98f5680-36f1-11eb-8d68-f32f32d9f60e.png)

Yes, the exception will if i `mvn clean package` with java11 and run with java11, so i change the java11 to java8, it work.

**Comment by supermoonie** [comment]
@FriwiDev 
Yes, the exception will if i `mvn clean package` with java11 and run with java11, so i change the java11 to java8, it work.

![image](https://user-images.githubusercontent.com/17509860/101233346-3b03f500-36f3-11eb-8662-189b072a9c50.png)

**Comment by FriwiDev** [comment]
The MacOS app bundle uses Java 8. I don't think that I can change that.

**Comment by rockwotj** [comment]
> The MacOS app bundle uses Java 8. I don't think that I can change that.

I'm not sure that is the case, the latest versions seems to use java 11.

https://github.com/jcefbuild/jcefbuild/issues/6#issuecomment-686729405

**Comment by fabio1999ita** [comment]
is possible use maven in old version too?

**Comment by FriwiDev** [comment]
Old java version (>8) yes, old jcef/cef version only if you import it into a private maven repo.

**Comment by fabio1999ita** [comment]
> Old java version (>8) yes, old jcef/cef version only if you import it into a private maven repo.

yeah, i talk about old jcef, like i remember you repo for v84.3 but it doesn't work

```xml
<repository>
      <id>jcef-maven</id>
      <name>JCef Maven Repository</name>
      <url>https://friwi.me/jcef/maven</url>
</repository>
```

```xml
<dependency>
      <groupId>org.cef</groupId>
      <artifactId>jcef</artifactId>
      <version>84.3.8+gc8a556f+chromium-84.0.4147.105-build-17</version>
</dependency>
<dependency>
      <groupId>org.cef</groupId>
      <artifactId>jcef-natives-win64</artifactId>
      <version>84.3.8+gc8a556f+chromium-84.0.4147.105</version>
</dependency>
```

**Comment by FriwiDev** [comment]
I dont know if this still works. Anyways, it is not recommended to still create flash applications in 2022. Look at an old commit in the jcefmaven repo to find usage instructions for the old version.
